### PR TITLE
feat: recursive subqueries

### PR DESCRIPTION
**Pull request by QuantumExplorer**
This allows for recursively applying queries to tree elements returned by parent queries.

### CODE REVIEW

**Review by fominok — APPROVED** [review]

File: grovedb/src/subtree.rs [hunk 1]
```diff
@@ -109,51 +112,63 @@ impl Element {
...
-
-                    if let Some(subquery) = subquery {
+                if subtrees_option.is_none() && subquery.is_none() {
+                    return Err(Error::InvalidPath(
+                        "you must provide a subquery or a subquery_key when interacting with a \
+                         tree of trees",
+                    ));
+                }
+                let subtrees = subtrees_option.ok_or(Error::MissingParameter(
+                    "subtrees must be provided when using a subquery key",
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply grovedb/src/subtree.rs:122]: Thats a good idea.

File: grovedb/src/subtree.rs [hunk 2]
```diff
@@ -109,51 +112,63 @@ impl Element {
...
-                        ))?
-                        .to_vec();
-                    path_vec.push(key.ok_or(Error::MissingParameter(
-                        "the key must be provided when using a subquery key",
-                    ))?);
-
-                    if let Some(subquery) = subquery {
+                if subtrees_option.is_none() && subquery.is_none() {
+                    return Err(Error::InvalidPath(
+                        "you must provide a subquery or a subquery_key when interacting with a \
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply grovedb/src/subtree.rs:117]: Fixed

File: merk/src/proofs/query/mod.rs [hunk 3]
```diff
@@ -175,7 +190,11 @@ impl Query {
 impl<Q: Into<QueryItem>> From<Vec<Q>> for Query {
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply merk/src/proofs/query/mod.rs:186]: I'll let you make a PR :D

File: grovedb/src/subtree.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply grovedb/src/subtree.rs:121]: It's because by definition if you are on a tree you are doing a subquery

**Pull request merged**